pkg/build/controller/controller.go

```diff
@@ -201,8 +201,11 @@ func (bc *BuildController) nextBuildPhase(build *buildapi.Build) error {
 
 	if _, err := bc.PodManager.CreatePod(build.Namespace, podSpec); err != nil {
 		if errors.IsAlreadyExists(err) {
-			bc.Recorder.Eventf(build, kapi.EventTypeWarning, "failedCreate", "Pod already exists: %s/%s", podSpec.Namespace, podSpec.Name)
+			bc.Recorder.Eventf(build, kapi.EventTypeWarning, "FailedCreate", "Pod already exists: %s/%s", podSpec.Namespace, podSpec.Name)
 			glog.V(4).Infof("Build pod already existed: %#v", podSpec)
+			build.Status.Phase = buildapi.BuildPhaseError
+			build.Status.Reason = buildapi.StatusReasonBuildPodExists
+			build.Status.Message = err.Error()
```

Comment:
We need to be checking if the pod is older than the build.  if the pod is newer than the build, we should continue to no-op this (it indicates something else managed to create the pod for us).